pkg/kratos/service.go

```diff
@@ -120,6 +138,65 @@ func (s *Service) GetLoginFlow(ctx context.Context, id string, cookies []*http.C
 	return flow, resp.Cookies(), nil
 }
 
+func (s *Service) GetRecoveryFlow(ctx context.Context, id string, cookies []*http.Cookie) (*kClient.RecoveryFlow, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.FrontendApi.GetRecoveryFlow")
+	defer span.End()
+
+	flow, resp, err := s.kratos.FrontendApi().
+		GetRecoveryFlow(ctx).
+		Id(id).
+		Cookie(cookiesToString(cookies)).
+		Execute()
+	if err != nil {
+		s.logger.Debugf("full HTTP response: %v", resp)
+		return nil, nil, err
+	}
+
+	s.logger.Debugf("Get recovery flow: %s", flow)
+
+	return flow, resp.Cookies(), nil
+}
+
+func (s *Service) UpdateRecoveryFlow(
+	ctx context.Context, flow string, body kClient.UpdateRecoveryFlowBody, cookies []*http.Cookie,
+) (*BrowserLocationChangeRequired, []*http.Cookie, error) {
+	ctx, span := s.tracer.Start(ctx, "kratos.FrontendApi.UpdateRecoveryFlow")
+	defer span.End()
+
+	_, resp, err := s.kratos.FrontendApi().
+		UpdateRecoveryFlow(ctx).
+		Flow(flow).
+		UpdateRecoveryFlowBody(body).
+		Cookie(cookiesToString(cookies)).
+		Execute()
+
+	// We expect to get a 422 response from Kratos. The sdk forces us to
+	// make the request with an 'application/json' content-type, whereas Kratos
+	// expects the 'Content-Type' and 'Accept' to be 'application/x-www-form-urlencoded'.
+	// This is not a real error, as we still get the URL to which the user needs to be
+	// redirected to.
+	if err != nil && resp.StatusCode != 422 {
```

Comment:
question: so this assumes that if `StatusCode` is actually 422 the `err` cannot be `nil`, meaning: this two conditions will happen together every time. Is this always the case?
If this is the case than we're good. If this is not the case than we would need an extra check on err

Reply:
Yes, if the status code is 422, we always get a `browser location change required` error which is not a real one (as stated in the comment), it's just an indication that we need to change the redirect url